react/class_components/component_lifecycle_methods.md

```diff
@@ -0,0 +1,75 @@
+
+In this method you should be updating anything that needs to be changed in response to either the DOM changing, or any states that you might want to act on upon change. For example, you'd want to refetch user data if the user changes.
+
+### componentWillUnmount()
+
+This is the last lifecycle method, which is called before a component is unmounted and destroyed. In this method you should be performing cleanup actions, so that would be cancelling network requests, clearing timers, etc.
+
+### useEffect()
+
+Now that we've learnt about class lifecycle methods, it's useful to understand that the `useEffect` hook used in functional components is essentially a combination of componentDidMount, componentDidUpdate, and componentWillUnmount. Which method/methods it relates to varies based on it's dependency array, and if it returns anything.
+
+*   An empty dependency array would be equivalent to `componentDidMount`.
+*   A dependency array with a value/values in it would be similar to `componentDidUpdate`, but only ran when the dependency provided changes.
+*   No dependency array would be equivalent to `componentDidMount` and `componentDidUpdate` combined.
+*   A return function inside of a `useEffect()` hook would be equivalent to `componentWillUnmount`.
+
+For example:
+
+~~~js
+    useEffect(() => {
+        placeholderFunction()
+        return () => clearUpFunction()
+    }, [])
+~~~
+
+In this snippet, the useEffect contains the functionality of `componentDidMount`, and `componentWillUnmount` via the return function.
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+1.  Check out this [lifecycle diagram](https://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/) to see a good visual representation of a components lifecycle methods.
+
+2. Read this [component lifecycle documentation](https://reactjs.org/docs/react-component.html#the-component-lifecycle) by the people who made react!
+    *   You only have to worry about the methods in bold, but you can read up on the others if you're curious - they're uncommon and you likely won't need them in 
+        most of your apps. 
+</div>
+
+### Knowledge Check
+
+This section contains questions for you to check your understanding of this lesson on your own. If you’re having trouble answering a question, click it and review the material it links to.
+
+*   <a class="knowledge-check-link" href="#render">What is the only required lifecycle method?</a>
+*   <a class="knowledge-check-link" href="#componentdidmount">What lifecycle method should you use for initial data fetching?</a>
+*   <a class="knowledge-check-link" href="#componentdidupdate">When you want to act upon change of the DOM, or of state, what lifecycle method would you use?</a>
+*   <a class="knowledge-check-link" href="#componentwillunmount">When performing cleanup actions, what lifecycle method should be used?</a>
```

Comment:
We can add a knowledge check for the useEffect section. Something like, "How the useEffect hook combines some of the lifecycle methods?"